SimPEG/EM/NSEM/Utils/plotUtils.py

```diff
@@ -0,0 +1,879 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import properties
+from SimPEG.EM.Utils.EMUtils import mu_0, omega
+
+# Define the default component dictionaries
+DEFAULT_COMP_DICT = {
+    'xx':{'color': 'green', 'label': 'Imp_xx', 'marker': '_', 'ls': 'None'},
+    'xy':{'color': 'blue', 'label': 'Imp_xy', 'marker': '_', 'ls': 'None'},
+    'yx':{'color': 'red', 'label': 'Imp_yx', 'marker': '_', 'ls': 'None'},
+    'yy':{'color': 'yellow', 'label': 'Imp_yy', 'marker': '_', 'ls': 'None'},
+    'zx':{'color': 'brown', 'label': 'Tip_zx', 'marker': '_', 'ls': 'None'},
+    'zy':{'color': 'purple', 'label': 'Tip_zy', 'marker': '_', 'ls': 'None'},
+}
+
+# Define some hidden attributes
+_imp_comps = ['xx', 'xy', 'yx', 'yy']
+
+def _validate_kwargs(input_dict, compare_dict):
+    """
+    Function to deal with keyword arguments.
+
+    :param input_dict: matplotlib kwargs dictionary with custom arguments
+    :type input_dict: :class: `dict`
+    :param compare_dict: matplotlib kwargs of default to use arguments
+    :type compare_dict: :class: `dict`
+    """
+    # Set the default plot kwargs
+    for key, val in compare_dict.items():
+        # Make sure they aren't already assigned
+        if input_dict is None:
+            input_dict = compare_dict.copy()
+        else:
+            if key not in input_dict:
+                input_dict[key] = val
+    # Return
+    return input_dict
+
+
+class Base_DataNSEM_plots(properties.HasProperties):
+    """
+    A class container of matplotlib panels for plotting
+    NSEM data.
+
+    """
+    fig = properties.Instance(
+        'Figure plotting',
+        plt.Figure,
+        required=False)
+    axes = properties.List(
+        'List of plot axes',
+        properties.Instance('Axes to plot the on', plt.Axes),
+        required=False)
+
+    def setup(self):
+        """
+        Setup up the plot window.
+
+        Should populate the
+            self.fig and self.axes properties
+        """
+        raise NotImplementedError('Is required in subclasses of {}'.format(self.__class__))
+
+    def draw(self):
+        raise NotImplementedError('Is required in subclasses of {}'.format(self.__class__))
+
+
+    def clear_axes(self):
+        """
+        Function to clear all of the axes
+        """
+        for ax in self.axes:
+            while len(ax.lines) > 0:
+                for line in ax.lines:
+                    ax.lines.remove(line)
+
+            while len(ax.collections) > 0:
+                for item in ax.collections:
+                    item.remove()
+
+
+class tip_amp_station_plot(Base_DataNSEM_plots):
```

Comment:
Typically class names are `CamelCase` and don't have `_under_scores`. Up to you.